formula/src/main/java/com/instacart/formula/IFormula.kt

```diff
@@ -0,0 +1,18 @@
+package com.instacart.formula
+
+/**
+ * Represents a basic reactive program that takes an [Input] and returns an [Output] over time. Take
+ * a look at [Formula] for the implementation details.
+ *
+ * @param Input Data and callbacks that are provided by the parent.
+ * @param Output Data and callbacks that are returned by this formula.
+ */
+interface IFormula<Input, Output> {
+
+    /**
+     * The actual implementation is deferred.
```

Comment:
I think this could use a little more explanation about what the purpose is. Maybe mentioning the fact that you can override and use this to test.